osu.Game/Screens/OnlinePlay/Lounge/Components/RoomStatusPill.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Online.Rooms;
+using osu.Game.Online.Rooms.RoomStatuses;
+using osuTK.Graphics;
+
+namespace osu.Game.Screens.OnlinePlay.Lounge.Components
+{
+    /// <summary>
+    /// A pill that displays the room's current status.
+    /// </summary>
+    public class RoomStatusPill : OnlinePlayComposite
+    {
+        [Resolved]
+        private OsuColour colours { get; set; }
+
+        private bool firstDisplay = true;
+        private PillContainer pill;
+        private SpriteText statusText;
+
+        public RoomStatusPill()
+        {
+            AutoSizeAxes = Axes.Both;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            InternalChild = pill = new PillContainer
+            {
+                Child = statusText = new OsuSpriteText
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    Font = OsuFont.GetFont(weight: FontWeight.SemiBold, size: 12),
+                    Colour = Color4.Black
+                }
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            EndDate.BindValueChanged(_ => updateDisplay());
+            Status.BindValueChanged(_ => updateDisplay(), true);
+        }
+
+        private void updateDisplay()
+        {
+            RoomStatus status = EndDate.Value < DateTimeOffset.Now ? new RoomStatusEnded() : Status.Value ?? new RoomStatusOpen();
```

Comment:
This line is a bit much (ternary into null-coalesce). Would maybe split lines for the branches of the ternary or add some parens for clarification.

Reply:
Also not sure it should be allowed to be null in the first place. Maybe worthwhile to set a default value to avoid the need for the `??` logic?

Reply:
It shouldn't be possible to be null. This may have been legacy code and `Room` defines a default value. Removed.